http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/sender/channel/pool/ConnectionManager.java

```diff
@@ -185,9 +187,11 @@ public void executeTargetChannel(HttpRoute httpRoute, SourceHandler sourceHandle
                 targetHandler.setIncomingMsg(carbonMessage);
                 targetHandler.setConnectionManager(connectionManager);
                 targetHandler.setTargetChannel(targetChannel);
-                if (ChannelUtils.writeContent(targetChannel.getChannel(), httpRequest, carbonMessage)) {
-                    targetChannel.setRequestWritten(true); // If request written
-                }
+                int socketIdleTimeout = senderConfiguration.getSocketIdleTimeout(60000);
+                targetChannel.getChannel().pipeline().addBefore("targetHandler", "idleStateHandler",
```

Comment:
shall we have a constant for "idleStateHandler" as it is used in multiple locatoins?